includes/functions.php

```diff
@@ -6186,7 +6192,28 @@ function build_entry_popup ( $popupid, $user, $description = '', $time,
   } //if $description
   return $ret . ( empty ( $site_extras ) ? '' : $site_extras ) . "</dl>\n";
 }
+// Craig. Several places in the code call fopen(),fget(),fclose()
+// then strip php open/close and comments.
+// and trim()
+// This would replace them.
+/**
+ * Read a file into an array.
+ *
+ * param  string  name of file
+ * param  bool    strip comments or not
+ */
+function file2array ( $file, $nocomments = 0 ) {
+  $ret = [];
+  foreach ( file ( $file, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES ) as $l ) {
+    // Strip php open/close and comments?
+    if ( $nocomments && preg_match ( '/^\s*(<\?|#|\/\*|\*|\?>)/', $l ) )
+      continue;
```

Comment:
The idea seems good... might want to unit test the function to verify it works as expected.

Reply:
It worked in my tests. No problems.